tests/VirtoCommerce.CatalogModule.Tests/ProductAssociationTests.cs

```diff
@@ -0,0 +1,390 @@
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using VirtoCommerce.CatalogModule.Data.Repositories;
+using VirtoCommerce.CatalogModule.Data.Services;
+using Xunit;
+using dataModel = VirtoCommerce.CatalogModule.Data.Model;
+using MockQueryable.Moq;
+using VirtoCommerce.CatalogModule.Core.Model;
+using VirtoCommerce.Platform.Caching;
+using VirtoCommerce.Platform.Core.Domain;
+
+namespace VirtoCommerce.CatalogModule.Test
+{
+    [Trait("Category", "CI")]
+    public class ProductAssociationTests
+    {
+        Mock<ICatalogRepository> _catalogRepositoryMock = new Mock<ICatalogRepository>();
+
+        public static object[][] ValidEntities => new object[][]
+        {
+            new object[]
+            {
+                new dataModel.AssociationEntity()
+                {
+                    Id = "originalId",
+                    ItemId = "originalItemId",
+                    AssociatedItemId = "originalAssociatedId",
+                    AssociatedCategoryId = null,
+                    AssociationType = "product",
+                    Priority = 0
+                },
+            },
+        };
+
+        [Theory]
+        [MemberData(nameof(ValidEntities))]
+        public async Task AddNotTransientAssociation(dataModel.AssociationEntity entity)
+        {
+
+            // Arrange
+            var associationServiceMock = CreateProductAssociationServiceMock(new []{ entity});
+
+            var productAssociation = new ProductAssociation()
+            {
+                Id = entity.Id,
+                ItemId = "new_Item_ID",
+                AssociatedObjectId =  "new_object_Id",
+                AssociatedObjectType = entity.AssociationType,
+            };
+            // Act
+            
+            await associationServiceMock.UpdateAssociationsAsync(new[] {productAssociation});
+            // Assert
+            Assert.Equal(productAssociation.ItemId, entity.ItemId);
```

Comment:
Probably here you need to check that Add was called. And probably you are checking not adding here, but update entity with the same Id - so you need to ensure that Add was not called, and ItemId and AssociatedObjectId were changed.
Also probably the correct name of the method is `UpdateAssociationsAsync_UpdateNotTransientAssociation_Changed`